contracts/AdjudicationFramework/MinimalAdjudicationFramework.sol

```diff
@@ -0,0 +1,234 @@
+
+    uint32 public constant REALITY_ETH_TIMEOUT = 86400;
+    uint32 public constant REALITY_ETH_BOND_ARBITRATOR_REMOVE = 10000;
+    uint32 public constant REALITY_ETH_BOND_ARBITRATOR_FREEZE = 20000;
+
+    // Template used to remove arbitrators, in case they are misbehaving
+    uint256 public templateIdRemoveArbitrator;
+
+    // Contract used for requesting a fork in the L1 chain in remove propositions
+    address public forkArbitrator;
+
+    // Propositions to manage arbitrators
+    enum PropositionType {
+        NONE,
+        ADD_ARBITRATOR,
+        REMOVE_ARBITRATOR
+    }
+    // Reality.eth questions for propositions we may be asked to rule on
+    struct ArbitratorProposition {
+        PropositionType proposition_type;
+        address arbitratorToRemove;
+        address newArbitrator;
+        bool isFrozen;
+    }
+    mapping(bytes32 => ArbitratorProposition) public propositions;
+
+    // Keep a count of active propositions that freeze an arbitrator.
+    // When they're all cleared they can be unfrozen.
+    mapping(address => uint256) public countArbitratorFreezePropositions;
+
+    IRealityETH public realityETH;
+
+    modifier onlyArbitrator() {
+        if (!contains(msg.sender)) {
+            revert OnlyAllowlistedActor();
+        }
+        _;
+    }
+
+    /// @param _realityETH The reality.eth instance we adjudicate for
+    /// @param _forkArbitrator The arbitrator contract that escalates to an L1 fork, used for our governance
+    /// @param _initialArbitrators Arbitrator contracts we initially support
+    constructor(
+        address _realityETH,
+        address _forkArbitrator,
+        address[] memory _initialArbitrators
+    ) {
+        realityETH = IRealityETH(_realityETH);
+        forkArbitrator = _forkArbitrator;
+
+        // Create reality.eth templates for our add and remove questions
+        // We'll identify ourselves in the template so we only need a single parameter for questions, the arbitrator in question.
+        // TODO: We may want to specify a document with the terms that guide this decision here, rather than just leaving it implicit.
+        string
+            memory templatePrefixRemove = '{"title": "Should we remove arbitrator %s and replace them by %s from the framework ';
+        string
+            memory templateSuffix = '?", "type": "bool", "category": "adjudication", "lang": "en"}';
+
+        string memory thisContractStr = Strings.toHexString(address(this));
+        string memory removeTemplate = string.concat(
+            templatePrefixRemove,
+            thisContractStr,
+            templateSuffix
+        );
+        templateIdRemoveArbitrator = realityETH.createTemplate(removeTemplate);
+
+        // Allowlist the initial arbitrators
+        for (uint256 i = 0; i < _initialArbitrators.length; i++) {
+            _addToList(_initialArbitrators[i]);
+        }
+    }
+
+    function beginReplaceArbitratorFromAllowList(
+        address arbitratorToRemove,
```

Comment:
How do we wanna deal with the situation where many arbitrators turn bad at the the same time. Woundn't it be nice to replace them all at once with one big question?